packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/component_builder.rs

```diff
@@ -242,6 +229,58 @@ impl ComponentBuilder {
         }
     }
 
+    /// If component `component_idx` extended a prop using the `extend` attribute,
+    /// then add a child to the component corresponding to that prop.
+    ///
+    /// For example, if `<text name="$i"/>`, then we add a child corresponding to `$i.value`
+    /// in the case of `<p extend="$i.value" />` but not to the text component coming from a `$i.value`
+    /// that occurs outside the `extend` attribute.
```

Comment:
`name="$i"` -> `name="i"`. I do not understand this example. Did you mean `<textInput name="i" />` and `<text extend="$i" />`?

Reply:
Since neither `<text>` nor `<textInput>` have rendered children, I was trying to use an example `<p>` that has rendered children (given that the need for rendered children was the reason we did it this way). I'll make the example have both `<text>` and `<p>` extending from a `<textInput>`.